### PR TITLE
Support Want-Digest header with fixity check for NonRDFSource.

### PR DESCRIPTION
**Pull request by lsitu**
Fixes https://jira.duraspace.org/browse/FCREPO-2601

Add support for Want-Digest header with fixity check on NonRDFSource.

### CODE REVIEW

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/BasicCacheEntry.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
  */
 package org.fcrepo.kernel.modeshape.utils;
 
+import org.apache.http.impl.auth.UnsupportedDigestAlgorithmException;
```
Comment (birkland) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/BasicCacheEntry.java]: I believe this should be `org.fcrepo.kernel.api.exception.UnsupportedAlgorithmException`.

When I do:
`curl -H "Want-Digest: sha256" -I http://localhost:8080/rest/img.jpg`

I get a 500 error, caught by the wildcard exception mapper.  The exception is:
<pre>
DEBUG 16:53:31.225 (WildcardExceptionMapper) WildcardExceptionMapper intercepted exception:{}

org.apache.http.impl.auth.UnsupportedDigestAlgorithmException: Unsupported digest algorithm: sha256
        at org.fcrepo.kernel.modeshape.utils.BasicCacheEntry.checkFixity(BasicCacheEntry.java:111)
        at org.fcrepo.kernel.modeshape.FedoraBinaryImpl.checkFixity(FedoraBinaryImpl.java:392)
        at org.fcrepo.http.api.FedoraLdp.handleWantDigestHeader(FedoraLdp.java:802)
        at org.fcrepo.http.api.FedoraLdp.head(FedoraLdp.java:217)
</pre>

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java [hunk 2]
```diff
@@ -770,6 +789,22 @@ private String mintNewPid(final String slug) {
         return pid;
     }
 
+    private String handleWantDigestHeader(final FedoraBinary binary, final String wantDigest)
+            throws UnsupportedAlgorithmException {
+        // handle the Want-Digest header with fixity check
+        final Collection<String> preferredDigests = parseWantDigestHeader(wantDigest);
+        if (preferredDigests.isEmpty()) {
+            LOGGER.warn("Unsupported digest algorithm provided in 'Want-Digest' header: {}", wantDigest);
```
Comment (birkland) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java]: `warn` seems to be too high of a log level.  The exception will already be logged at the `DEBUG` level by the exception handler for `UnsupportedAlgorithmException`, so I think you can just remove this

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/BasicCacheEntry.java (hunk 1 above)
Comment (lsitu) [reply fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/BasicCacheEntry.java]: Good catch! Thanks. I've added a commit to change the exception to UnsupportedAlgorithmException with a test. See https://github.com/fcrepo4/fcrepo4/pull/1231/commits/c306e2f9d6cfa2f7241becbe7f29f1cd1ed6f824.

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java (hunk 2 above)
Comment (lsitu) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java]: Sure, the warn logging is removed. See commit c306e2f.

**Pull request merged**